src/Rules.hs

```diff
@@ -61,6 +61,10 @@ packageRules = do
         [ Rules.Compile.compilePackage readPackageDb
         , Rules.Library.buildPackageLibrary ]
 
+    let dynContexts = liftM3 Context [Stage1 ..] knownPackages [dynamic]
```

Comment:
I am not a fan of using contractions. Can we use `dynamic` throughout the code instead of `dyn`? This will be more consistent with the rest of Hadrian.